### PR TITLE
[egs] Finalize DeMask recipe

### PR DESCRIPTION
**Pull request by popcornell**
This PR adds the bash part of the recipe. 

Additionally :
- Fix model saving
- Add noise support to DeMask for less domain mismatch
- Better license notice generation

### CODE REVIEW

File: egs/demask/run.sh [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+if [[ $stage -le  0 ]]; then
+  echo "Stage 0: Downloading required Datasets"
+
+  #if ! test -e $librispeech_dir/train-clean-360; then
+   #  echo "Downloading LibriSpeech/train-clean-360 into $storage_dir"
+    # wget -c --tries=0 --read-timeout=20 http://www.openslr.org/resources/12/train-clean-360.tar.gz -P $storage_dir
+	  # tar -xzf $storage_dir/dev-clean.tar.gz -C $storage_dir
+	  # rm -rf $storage_dir/dev-clean.tar.gz
+	#fi
```
Comment (mpariente) [inline egs/demask/run.sh:65]: Why is this commented

File: egs/demask/run.sh [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  echo "Stage 3: Training"
+  mkdir -p logs
+  CUDA_VISIBLE_DEVICES=$id $python_path train.py \
+		--clean_speech_train $dumpdir/clean/train-clean-360.json \
+		--clean_speech_valid $dumpdir/clean/dev-clean.json \
+	  --rir_train $dumpdir/rirs/train.json \
+	  --rir_valid $dumpdir/rirs/validation.json \
+		--fs $sample_rate \
+		--exp_dir ${expdir}/ | tee logs/train_${tag}.log
+	cp logs/train_${tag}.log $expdir/train.log
```
Comment (mpariente) [inline egs/demask/run.sh:113]: Indent this consistently please?

File: egs/demask/local/demask_dataset.py [hunk 3]
```diff
@@ -173,5 +214,15 @@ def get_infos(self):
...
+
+
+demask_license = dict(
+    title="Acoustic effects of medical, cloth, and transparent face masks on speech signals",
+    title_link="https://arxiv.org/abs/2008.04521",
+    author="Corey, Ryan M and Jones, Uriah and Singer, Andrew C",
+    license="CC BY-NC 4.0",
+    license_link="https://creativecommons.org/licenses/by-nc/4.0/",
+    non_commercial=True,
+)
```
Comment (mpariente) [inline egs/demask/local/demask_dataset.py:228]:
```suggestion
demask_license = dict(
    title="Acoustic effects of medical, cloth, and transparent face masks on speech signals",
    title_link="https://arxiv.org/abs/2008.04521",
    author="Corey, Ryan M and Jones, Uriah and Singer, Andrew C",
    license="CC BY 4.0",
    license_link="https://creativecommons.org/licenses/by/4.0/",
    non_commercial=False,
)
```

**Review by mpariente — APPROVED** [review]
Looks good, almost ready to merge

File: egs/demask/run.sh (hunk 1 above)
Comment (popcornell) [reply egs/demask/run.sh:65]: whoops

File: egs/demask/run.sh [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+# Path to the python you'll use for the experiment. Defaults to the current python
+# You can run ./utils/prepare_python_env.sh to create a suitable python environment, paste the output here.
+python_path=python
+
+# Example usage
+# ./run.sh --stage 3 --tag my_tag --task sep_noisy --id 0,1
+
+# General
+stage=3  # Controls from which stage to start
```
Comment (mpariente) [inline egs/demask/run.sh:23]:
```suggestion
stage=0
```

**Comment by faroit** [comment]
just in time ;-)

**Pull request merged**